### PR TITLE
fix(fetch): prevent body stringify when `FormData`

### PR DESCRIPTION
**Pull request by benjamincanac**

### CODE REVIEW

File: src/fetch.ts [hunk 1]
```diff
@@ -77,7 +77,9 @@ export function createFetch ({ fetch }: CreateFetchOptions): $Fetch {
         request = withQuery(request, opts.params)
       }
       const hasPayload = payloadMethods.includes((opts.method || '').toLowerCase())
-      if (opts.body && typeof opts.body === 'object' && hasPayload) {
+      const supportsFormData = typeof FormData === 'function'
+      const isFormData = supportsFormData && opts.body instanceof FormData
```
Comment (pi0) [inline src/fetch.ts:81]: We cannot rely on global FormData detection. One can directly use a package like [form-data](https://www.npmjs.com/package/form-data) in Node.js without global polyfill.

File: src/fetch.ts [hunk 2]
```diff
@@ -77,7 +77,9 @@ export function createFetch ({ fetch }: CreateFetchOptions): $Fetch {
...
+      if (opts.body && typeof opts.body === 'object' && !isFormData && hasPayload) {
```
Comment (danielroe) [inline src/fetch.ts:82]: how about something like this instead:
```suggestion
      if (opts.body && typeof opts.body === 'object' && hasPayload && opts.body.toString() !== '[object FormData]') {
```

Comment (pi0) [reply src/fetch.ts:82]: Pushed a generic fix but seems a good idea for using `[object FormData]` to append proper content-type (adding in short)

**Comment by pi0** [comment]
This issue should be fixed on 0.4.9

**Pull request closed without merging**